index.js

```diff
@@ -343,163 +341,159 @@ Filter.prototype._handleFile = function(relativePath, srcDir, destDir, entry, ou
+    instrumentation.processString++;
+    let processStringStart = process.hrtime();
+    let string = invoke(this.processor, this.processor.processString, [this, contents, relativePath, forceInvalidation, instrumentation]);
 
-  instrumentation.processString++;
-  let processStringStart = process.hrtime();
-  let string = invoke(this.processor, this.processor.processString, [this, contents, relativePath, forceInvalidation, instrumentation]);
+    return string.then(outputString => {
+      instrumentation.processStringTime += nanosecondsSince(processStringStart);
+      let outputPath = filter.getDestFilePath(relativePath, entry);
 
-  return string.then(outputString => {
-    instrumentation.processStringTime += nanosecondsSince(processStringStart);
-    let outputPath = filter.getDestFilePath(relativePath, entry);
+      if (outputPath == null) {
+        throw new Error('canProcessFile("' + relativePath +
+                        '") is true, but getDestFilePath("' +
+                        relativePath + '") is null');
+      }
 
-    if (outputPath == null) {
-      throw new Error('canProcessFile("' + relativePath +
-                      '") is true, but getDestFilePath("' +
-                      relativePath + '") is null');
-    }
+      outputPath = destDir + '/' + outputPath;
 
-    outputPath = destDir + '/' + outputPath;
+      if (isChange) {
+        let isSame = fs.readFileSync(outputPath, 'UTF-8') === outputString;
+        if (isSame) {
+          this._logger.debug('[change:%s] but was the same, skipping', relativePath, isSame);
+          return;
+        } else {
+          this._logger.debug('[change:%s] but was NOT the same, writing new file', relativePath);
+        }
+      }
 
-    if (isChange) {
-      let isSame = fs.readFileSync(outputPath, 'UTF-8') === outputString;
-      if (isSame) {
-        this._logger.debug('[change:%s] but was the same, skipping', relativePath, isSame);
-        return;
-      } else {
-        this._logger.debug('[change:%s] but was NOT the same, writing new file', relativePath);
+      try {
+        fs.writeFileSync(outputPath, outputString, {
+          encoding: outputEncoding
+        });
+
+      } catch(e) {
+        // optimistically assume the DIR was patched correctly
+        mkdirp.sync(path.dirname(outputPath));
+        fs.writeFileSync(outputPath, outputString, {
+          encoding: outputEncoding
+        });
       }
-    }
 
-    try {
-      fs.writeFileSync(outputPath, outputString, {
-        encoding: outputEncoding
-      });
+      return outputString;
+    });
+  }
 
-    } catch(e) {
-      // optimistically assume the DIR was patched correctly
-      mkdirp.sync(path.dirname(outputPath));
-      fs.writeFileSync(outputPath, outputString, {
-        encoding: outputEncoding
-      });
-    }
+  /**
+   * @param contents {string}
+   * @param relativePath {string}
+   * @returns {string}
+   */
+  processString (contents, relativePath) { // jshint ignore:line
```

Comment:
do we need this jshint comment?

Reply:
Yeah, we need as we are not using those variables inside function, JShint will throw an error.

Reply:
i think you can prefix them with `_` to make jshint ignore them

Reply:
We could then the signature of the line changes we may have to change the `@param` above it.

Reply:
https://github.com/stefanpenner/broccoli-persistent-filter/blob/1b78094d3eea0e70a51600e9bc1133443c45579c/index.js#L482-L483

Reply:
sure

Reply:
https://ci.appveyor.com/project/embercli/broccoli-persistent-filter/builds/29323823/job/dr1buhh5asmaqwka. Fails looks like we have to add the jshint igonre.